dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -436,6 +437,25 @@ internal class DatadogRumMonitor(
         }
     }
 
+    /**
+     * Wait for any pending events. This is mostly for integration tests to ensure that the
+     * RUM context is in the correct state before proceeding.
+     */
+    @Suppress("unused")
+    private fun waitForPendingEvents() {
+        if (!executorService.isShutdown) {
+            val latch = CountDownLatch(1)
+            // Submit an empty task, and wait for it to complete
+            executorService.submit {
+                latch.countDown()
+            }
+            try {
+                latch.await()
```

Comment:
Maybe we could add a timeout here in case something is blocking the executor indefinitely?